### DISCUSSION TITLE
How do you get stylelint --fix to work with styled-jsx?

### DISCUSSION THREAD

**Discussion by etler** [vercel/next.js#18323] [Help]
I'm trying to figure out how to get stylelint to play nicely with styled-jsx. I first tried out the postcss plugin, but could not figure out when it was supposed to run, it seems like it makes it a part of the build process? But I did not see errors on build. Also, we  want to be able to run the linter outside of the build process at any time so I'm not sure if the plugin approach allowed for that?

Since I couldn't get that to work I followed [this guide](https://github.com/vercel/styled-jsx/issues/106#issuecomment-304395750) instead, and that worked great! But the problem was due to it requiring a processor, it broke support for `stylelint --fix` as [that does not seem to support processors](https://github.com/styled-components/stylelint-processor-styled-components/issues/264).

So it seems the two options I have are to either remove the need for a processor via the configuration solution I'm using, or to get the plugin approach to work, but I'm not actually sure if that approach supports `--fix` and I'm not sure if it's possible to use that approach for linting outside of build time.

Has anyone gotten `--fix` to work with styled-jsx? Thanks for any help!

**Comment by ryan-lustre** [comment]
stylelint supports syntaxes very close to styled-jsx out of the box via [@stylelint/postcss-css-in-js][] (we haven’t run into any specific incompatibilities yet, I think?), so `--fix` will work after removing both the PostCSS plugin and the processor.

[@stylelint/postcss-css-in-js]: https://github.com/stylelint/postcss-css-in-js
